apps/bam.iobio/src/App.tsx

```diff
@@ -118,26 +115,46 @@ function App() {
 			</button>
 
 			<div className={clsx('bam-container', showBam && 'bam-open')}>
-				{/* Needs to render on the page before scripts for BAM to work */}
 				<div id="app"></div>
 				{showBam ? (
 					<>
 						<h3>Bam.Iobio</h3>
 
-						<iobio-data-broker alignment-url="https://s3.amazonaws.com/iobio/NA12878/NA12878.autsome.bam" />
+						<IobioDataBroker url={iobioURL} />
 
+						{/* Percent Boxes */}
 						<div className="row iobio-container">
-							{mappedReads && <iobio-percent-box percent-key="mapped_reads" total-key="total_reads" />}
-							{forwardStrands && <iobio-percent-box percent-key="forward_strands" total-key="total_reads" />}
-							{properPairs && <iobio-percent-box percent-key="proper_pairs" total-key="total_reads" />}
-							{singletons && <iobio-percent-box percent-key="singletons" total-key="total_reads" />}
-							{bothMatesMapped && <iobio-percent-box percent-key="both_mates_mapped" total-key="total_reads" />}
-							{duplicates && <iobio-percent-box percent-key="duplicates" total-key="total_reads" />}
-						</div>
-						<div className="row iobio-histo-container">
-							{histogram && <iobio-histogram broker-key="coverage_hist" />}
-							{/* {histogram && <Histogram broker-key="coverage_hist" />} */}
+							{percentKeys.map(
+								(key) =>
+									bamContext[key] && (
+										<IobioPercentBox title={BamDisplayNames[key]} percentKey={key} totalKey="total_reads" key={key} />
+									),
+							)}
 						</div>
+
+						{/* Coverage Depth */}
+						{bamContext.coverage_depth && (
+							<div className="row iobio-chart-container">
+								<IobioCoverageDepth />
+							</div>
+						)}
+
+						{/* Histograms */}
+						{histogramKeys.map(
+							(key, index) =>
+								bamContext[key] && (
+									<div key={key} className="row iobio-chart-container">
+										<IobioHistogram
+											key={key}
+											brokerKey={key}
+											title={BamDisplayNames[key]}
+											color={colors[index]}
+											// TODO: why not boolean false?
+											ignoreOutliers={ignoreOutlierKeys.includes(key) || undefined}
```

Comment:
not sure the need for `undefined`
type is optional boolean, so you can pass it the `ignoreOutlierKeys.includes(key)` part of just leave the prop out.

Reply:
The problem is that the charts render incorrectly when the boolean is false but render correctly when this value is undefined
This points to an issue with `iobio-charts` rather than my code which is why I marked it as TODO
Will revisit

Reply:
Gotcha. So it's a stop gap solution until charts is fixed?
One suggestion would be to keep these fixes constrained to the React components. 
So it's a single place to fix it later instead of switching out the usage of the React components.
(general idea, not just this point)

Reply:
Chatted w/ Anders yesterday, ignoreOutliers is just using standard html boolean attributes , i.e. looking for the presence or absence of this property
https://developer.mozilla.org/en-US/docs/Glossary/Boolean/HTML

Updated here: [update ignoreOutliers](https://github.com/overture-stack/iobio-components/pull/24/commits/af65f19ee48a515799c8778cb4a4db2a69a876f2)

Reply:
I see that now -- updating in connection with Type updates

Reply:
Better: [Improved Types + Outlier handling](https://github.com/overture-stack/iobio-components/pull/24/commits/2fb3929436379660d3f94c780c7e1bf029f2292e)

Reply:
I've separated some of the Types/Constants into the `components` package because they represent inherent properties related to the Bam Components

And then separated the rest into the Demo App util file, because they are more related to page layout and may vary depending on the specific implementation

[Move Types / Constants](https://github.com/overture-stack/iobio-components/pull/24/commits/5a010ae493f2f28cb5a92cbd8e9ed4b4af101735)